### PR TITLE
VIP: Migrate imports and exports from CommonJS to ES2015

### PR DESCRIPTION
**Pull request by jsnmoon**
Part of #11688. 

This change will allow us to leverage [Webpack 2's tree shaking optimization for ES6 modules](https://gist.github.com/sokra/27b24881210b56bbaff7#es6-specific-optimizations) in the future. Please share your suggestions and comments!

### CODE REVIEW

**Comment by ockham** [comment]
Looks sane, tho you might have to be careful with the `export default`s -- those sometimes tend to break when imported CommonJS style.

But TBH, I don't know how to test VIP related stuff -- my changes to those files were part of a bulk change, and I'm not sure I even tested them myself or just asked @scottsweb to do that for me 😄 

BTW @scottsweb -- is this VIP stuff inside Calypso being used at all?

File: client/vip/controller.js [hunk 1]
```diff
@@ -1,19 +1,19 @@
...
+import i18n from 'i18n-calypso';
+import page from 'page';
 
 /**
  * Internal Dependencies
  */
-var setTitle = require( 'state/document-head/actions' ).setDocumentHeadTitle,
-	sites = require( 'lib/sites-list' )();
+import { setDocumentHeadTitle as setTitle } from 'state/document-head/actions';
+const sites = require( 'lib/sites-list' )();
```
Comment (spen) [inline client/vip/controller.js]: A pattern that's used in a few places for imports like `sites-list` looks something like this:
```js
import sitesFactory from 'lib/sites-list';
const sites = sitesFactory();
```
Though the naming isn't entirely consistent throughout the code base ([here](https://github.com/Automattic/wp-calypso/blob/master/client/my-sites/themes/themes-site-selector-modal.jsx), [here](https://github.com/Automattic/wp-calypso/blob/master/client/components/site-selector/add-site.jsx) & [here](https://github.com/Automattic/wp-calypso/blob/master/client/components/sites-dropdown/index.jsx) for examples)
Might it be worth making that change in this PR or leaving for another PR that targets this pattern specifically?

File: client/vip/controller.js [hunk 2]
```diff
@@ -1,19 +1,19 @@
...
+import React from 'react';
+import i18n from 'i18n-calypso';
+import page from 'page';
 
 /**
  * Internal Dependencies
  */
-var setTitle = require( 'state/document-head/actions' ).setDocumentHeadTitle,
-	sites = require( 'lib/sites-list' )();
+import { setDocumentHeadTitle as setTitle } from 'state/document-head/actions';
```
Comment (spen) [inline client/vip/controller.js]: Perhaps there's value in keeping `setDocumentHeadTitle` as-is vs aliasing (though I do see that either way it's out of the scope of this PR).

File: client/vip/controller.js (hunk 1 above)
Comment (jsnmoon) [inline client/vip/controller.js]: Hmm... let's punt that change for the next bulk change.

File: client/vip/controller.js (hunk 2 above)
Comment (jsnmoon) [inline client/vip/controller.js]: I'm personally on the same page, but I refrained from making too many discretionary edits :)

**Comment by ockham** [comment]
Hmm, looks like we can close this since `client/vip` was actually unused and recently removed (#11818) ¯\_(ツ)_/¯

**Comment by jsnmoon** [comment]
Haha, works for me!

**Pull request closed without merging**